zproject/prod_settings_template.py

```diff
@@ -458,6 +462,17 @@
     "emailAddress": ZULIP_ADMINISTRATOR,
 }
 
+# SOCIAL_AUTH_SYNC_CUSTOM_ATTRS_DICT = {
+#    "example_org": {
+#        "saml": {
+#            # The format is
+#            # <custom profile field name>: <attribute name from the auth provider to sync it with>
+#            "title": "title",
+#            "phone_number": "mobilePhone",
+#        }
+#    }
+# }
```

Comment:
I'm thinking about the design of having this be an independent block from the main SAML configuration.  I guess the benefit is that we could use this to populate attributes from GitHub/GitLab auth as well, which we may down the line want to do (E.g. to store your GitHub username somewhere when you auth with GitHub).  And the downside is that it's a second place to configure when setting up SAML, and we need to hardcode the mapping to the subdomain and SAML backend here.  I think I'm fine with this interface for now; we may want to think in the medium term about whether we want to move this phase of configuration to something that lives in the database.

Should the `saml` here be `saml:idp_name`, to support the possible configuration where we have multiple SAML backends associated with a single organization?

Reply:
Yup, the main intent here was to make this extensible to other backends. Agreed that it may be best to just move this whole thing into the database at some point, which would let us expose it in Org settings      instead of needing folks to tell us the mapping to put it `settings.py` on Zulip Cloud

Reply:
> Should the `saml` here be `saml:idp_name`, to support the possible configuration where we have multiple SAML backends associated with a single organization?

Do we want/need such a granular level of control? Presumably, the admin should have control over their SAML IdP configs and, if they're using multiple IdPs with Zulip, just make the naming of attrs consistent between them.

To support `saml:idp_name` would add a small bit of complexity because we'd need to plumb the idp_name into the social pipeline too - right now we only access `backend.name` which is simpler.

Reply:
Hmm, I was thinking about the granular reference mainly for Zulip Cloud use.  Maybe worth having a brief discussion on chat.zulip.org to get others' thoughts, but I'm OK merging as-is and tweaking in a near-term follow-up commit if needed.

Reply:
@timabbott I'm not sure why it would be needed for Zulip Cloud either? Just to clarify - this already supports per-organization mappings (the setting is a dict mapping a subdomain to a dict configuring mapping for each backend), so two different organization can have whatever Attribute names they want. It's only if one org wanted to have two separate SAML providers and sync custom fields from them and couldn't make the attribute naming consistent between them - which I think sounds rather unlikely?

Reply:
Yeah, OK, I think I agree this is correct as written. Thanks!